### ISSUE TITLE
Increment operators do not yield a value

### ISSUE THREAD

**Issue by SeidChr** [PowerShell/PowerShell#17136]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Coming from C#, I expect the ++ operator to yield the value of the variable that has been or will be incremented. This does not work on PowerShell (tested on 7.2.2)

`$x = 1; $x++` or `$x = 1; ++$x` do not yield a value

if I put round brackets around the expression, it works:
`$x = 1; ($x++)` -> 1
`$x = 1; (++$x)` -> 2

the operator should yield the respective value without the need to put brackets around it.

### Expected behavior

```console
PS> $x = 1; $x++
1

PS> $x = 1; ++$x
2
```

### Actual behavior

```console
PS> $x = 1; $x++
<nothing>

PS> $x = 1; ++$x
<nothing>
```

### Environment data

```powershell
$ $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.2
PSEdition                      Core
GitCommitId                    7.2.2
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by iRon7** [comment]
This is by design. PowerShell differs from C# where it has a [Expression Mode](https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_parsing#expression-mode) *and* an [argument mode](https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_parsing#argument-mode). Technically, "*yielding a value*" means that you send the value to the pipeline and eventually display it at the end of a pipeline (see: [`Write-Output`](https://docs.microsoft.com/powershell/module/microsoft.powershell.utility/write-output#description) which describes the default behavior).

In a nutshell:. PowerShell doesn't require parentheses when calling a function and in case you use  `$x++` as an argument, you do not want to put that on the pipeline when invoking a function like this:
```PowerShell
$Object |ConvertTo-Json -Depth $x++
```

**Comment by jhoneill** [comment]
This is intentional behaviour and it looks weird if you are not expecting it. 

```
($x = 2) -gt 1 
```
Returns true, because the value of an assignment operation is the value that was assigned, 
But `$x = 2` is _normally_ silent   ; so are `$x = $x +1`  `$x += 1` and `$x ++`. 

The example above needs brackets because otherwise it reads as `$x = (2 -gt 1)`  but something like  `++ $x  -gt 2`  will return true and doesn't need the brackets. 

This way you can write something like 
```
while ($x ++ -lt 10) do { "something with $x"} 
``` 
And the pre-increment value is available but  if you write the same thing as  
```
while ($x -lt 10) do { "something with $x" ; $x++} 
```
with the incremented value being sent as output.

**Comment by jhoneill** [comment]
Those tricks allow things like this 
```
> $z = $y = $x = 1  # Sets all 3 to 1 

> $z += $y += ++ $x  # Makes x 2, adds it y, and then adds Y to Z

> $x,$y,$z

2
3
4
```

**Comment by SeidChr** [comment]
thanks for your answers. guess ill need to check in on that expression vs. argument mode!
definitely something you need to get used to.
